// @ariakit/react-core@0.4.18/cjs/__chunks/4R2KFV2F.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _ZVJRPAXYcjs = require('./ZVJRPAXY.cjs');



var _D6FV6EYScjs = require('./D6FV6EYS.cjs');


var _WULEED4Qcjs = require('./WULEED4Q.cjs');

// src/select/select-context.tsx
var _react = require('react');
var ctx = _WULEED4Qcjs.createStoreContext.call(void 0, 
  [_ZVJRPAXYcjs.PopoverContextProvider, _D6FV6EYScjs.CompositeContextProvider],
  [_ZVJRPAXYcjs.PopoverScopedContextProvider, _D6FV6EYScjs.CompositeScopedContextProvider]
);
var useSelectContext = ctx.useContext;
var useSelectScopedContext = ctx.useScopedContext;
var useSelectProviderContext = ctx.useProviderContext;
var SelectContextProvider = ctx.ContextProvider;
var SelectScopedContextProvider = ctx.ScopedContextProvider;
var SelectItemCheckedContext = _react.createContext.call(void 0, false);
var SelectHeadingContext = _react.createContext.call(void 0, null);









exports.useSelectContext = useSelectContext; exports.useSelectScopedContext = useSelectScopedContext; exports.useSelectProviderContext = useSelectProviderContext; exports.SelectContextProvider = SelectContextProvider; exports.SelectScopedContextProvider = SelectScopedContextProvider; exports.SelectItemCheckedContext = SelectItemCheckedContext; exports.SelectHeadingContext = SelectHeadingContext;
