// @ariakit/react-core@0.4.18/cjs/__chunks/6IDGXS7Y.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _26P4PLHIcjs = require('./26P4PLHI.cjs');


var _VIIJ4PIKcjs = require('./VIIJ4PIK.cjs');


var _6PUNKVPOcjs = require('./6PUNKVPO.cjs');


var _FCZ3IGUBcjs = require('./FCZ3IGUB.cjs');


var _25BPIGZHcjs = require('./25BPIGZH.cjs');


var _OZM4QA2Vcjs = require('./OZM4QA2V.cjs');



var _7EQBAZ46cjs = require('./7EQBAZ46.cjs');

// src/dialog/dialog-backdrop.tsx
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
function DialogBackdrop({
  store,
  backdrop,
  alwaysVisible,
  hidden
}) {
  const ref = _react.useRef.call(void 0, null);
  const disclosure = _6PUNKVPOcjs.useDisclosureStore.call(void 0, { disclosure: store });
  const contentElement = _25BPIGZHcjs.useStoreState.call(void 0, store, "contentElement");
  _react.useEffect.call(void 0, () => {
    const backdrop2 = ref.current;
    const dialog = contentElement;
    if (!backdrop2) return;
    if (!dialog) return;
    backdrop2.style.zIndex = getComputedStyle(dialog).zIndex;
  }, [contentElement]);
  _OZM4QA2Vcjs.useSafeLayoutEffect.call(void 0, () => {
    const id = contentElement == null ? void 0 : contentElement.id;
    if (!id) return;
    const backdrop2 = ref.current;
    if (!backdrop2) return;
    return _26P4PLHIcjs.markAncestor.call(void 0, backdrop2, id);
  }, [contentElement]);
  const props = _VIIJ4PIKcjs.useDisclosureContent.call(void 0, {
    ref,
    store: disclosure,
    role: "presentation",
    "data-backdrop": (contentElement == null ? void 0 : contentElement.id) || "",
    alwaysVisible,
    hidden: hidden != null ? hidden : void 0,
    style: {
      position: "fixed",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    }
  });
  if (!backdrop) return null;
  if (_react.isValidElement.call(void 0, backdrop)) {
    return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _FCZ3IGUBcjs.Role, _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {}, props), { render: backdrop }));
  }
  const Component = typeof backdrop !== "boolean" ? backdrop : "div";
  return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _FCZ3IGUBcjs.Role, _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {}, props), { render: /* @__PURE__ */ _jsxruntime.jsx.call(void 0, Component, {}) }));
}



exports.DialogBackdrop = DialogBackdrop;
