// @ariakit/react-core@0.4.18/cjs/__chunks/6M54DBBO.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});





var _VNVMHTCYcjs = require('./VNVMHTCY.cjs');



var _6F6SPO63cjs = require('./6F6SPO63.cjs');



var _D6FV6EYScjs = require('./D6FV6EYS.cjs');


var _WULEED4Qcjs = require('./WULEED4Q.cjs');

// src/menu/menu-context.tsx
var _react = require('react');
var menu = _WULEED4Qcjs.createStoreContext.call(void 0, 
  [_D6FV6EYScjs.CompositeContextProvider, _6F6SPO63cjs.HovercardContextProvider],
  [_D6FV6EYScjs.CompositeScopedContextProvider, _6F6SPO63cjs.HovercardScopedContextProvider]
);
var useMenuContext = menu.useContext;
var useMenuScopedContext = menu.useScopedContext;
var useMenuProviderContext = menu.useProviderContext;
var MenuContextProvider = menu.ContextProvider;
var MenuScopedContextProvider = menu.ScopedContextProvider;
var useMenuBarContext = _VNVMHTCYcjs.useMenubarContext;
var useMenuBarScopedContext = _VNVMHTCYcjs.useMenubarScopedContext;
var useMenuBarProviderContext = _VNVMHTCYcjs.useMenubarProviderContext;
var MenuBarContextProvider = _VNVMHTCYcjs.MenubarContextProvider;
var MenuBarScopedContextProvider = _VNVMHTCYcjs.MenubarScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);













exports.useMenuContext = useMenuContext; exports.useMenuScopedContext = useMenuScopedContext; exports.useMenuProviderContext = useMenuProviderContext; exports.MenuContextProvider = MenuContextProvider; exports.MenuScopedContextProvider = MenuScopedContextProvider; exports.useMenuBarContext = useMenuBarContext; exports.useMenuBarScopedContext = useMenuBarScopedContext; exports.useMenuBarProviderContext = useMenuBarProviderContext; exports.MenuBarContextProvider = MenuBarContextProvider; exports.MenuBarScopedContextProvider = MenuBarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
