// @ariakit/react-core@0.4.18/cjs/__chunks/D6FV6EYS.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _6HKL3JR2cjs = require('./6HKL3JR2.cjs');


var _WULEED4Qcjs = require('./WULEED4Q.cjs');

// src/composite/composite-context.tsx
var _react = require('react');
var ctx = _WULEED4Qcjs.createStoreContext.call(void 0, 
  [_6HKL3JR2cjs.CollectionContextProvider],
  [_6HKL3JR2cjs.CollectionScopedContextProvider]
);
var useCompositeContext = ctx.useContext;
var useCompositeScopedContext = ctx.useScopedContext;
var useCompositeProviderContext = ctx.useProviderContext;
var CompositeContextProvider = ctx.ContextProvider;
var CompositeScopedContextProvider = ctx.ScopedContextProvider;
var CompositeItemContext = _react.createContext.call(void 0, 
  void 0
);
var CompositeRowContext = _react.createContext.call(void 0, 
  void 0
);









exports.useCompositeContext = useCompositeContext; exports.useCompositeScopedContext = useCompositeScopedContext; exports.useCompositeProviderContext = useCompositeProviderContext; exports.CompositeContextProvider = CompositeContextProvider; exports.CompositeScopedContextProvider = CompositeScopedContextProvider; exports.CompositeItemContext = CompositeItemContext; exports.CompositeRowContext = CompositeRowContext;
