// @ariakit/react-core@0.4.18/cjs/__chunks/ESXJ577B.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _JLMLKTUBcjs = require('./JLMLKTUB.cjs');




var _WULEED4Qcjs = require('./WULEED4Q.cjs');



var _OZM4QA2Vcjs = require('./OZM4QA2V.cjs');



var _7EQBAZ46cjs = require('./7EQBAZ46.cjs');

// src/heading/heading.tsx
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
var TagName = "h1";
var useHeading = _WULEED4Qcjs.createHook.call(void 0, 
  function useHeading2(props) {
    const ref = _react.useRef.call(void 0, null);
    const level = _react.useContext.call(void 0, _JLMLKTUBcjs.HeadingContext) || 1;
    const Element = `h${level}`;
    const tagName = _OZM4QA2Vcjs.useTagName.call(void 0, ref, Element);
    const isNativeHeading = _react.useMemo.call(void 0, 
      () => !!tagName && /^h\d$/.test(tagName),
      [tagName]
    );
    props = _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {
      render: /* @__PURE__ */ _jsxruntime.jsx.call(void 0, Element, {}),
      role: !isNativeHeading ? "heading" : void 0,
      "aria-level": !isNativeHeading ? level : void 0
    }, props), {
      ref: _OZM4QA2Vcjs.useMergeRefs.call(void 0, ref, props.ref)
    });
    return props;
  }
);
var Heading = _WULEED4Qcjs.forwardRef.call(void 0, function Heading2(props) {
  const htmlProps = useHeading(props);
  return _WULEED4Qcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useHeading = useHeading; exports.Heading = Heading;
