// @ariakit/react-core@0.4.18/cjs/__chunks/I64JXNZF.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _D6FV6EYScjs = require('./D6FV6EYS.cjs');


var _WULEED4Qcjs = require('./WULEED4Q.cjs');

// src/tag/tag-context.tsx
var _react = require('react');
var TagValueContext = _react.createContext.call(void 0, null);
var TagRemoveIdContext = _react.createContext.call(void 0, 
  null
);
var ctx = _WULEED4Qcjs.createStoreContext.call(void 0, 
  [_D6FV6EYScjs.CompositeContextProvider],
  [_D6FV6EYScjs.CompositeScopedContextProvider]
);
var useTagContext = ctx.useContext;
var useTagScopedContext = ctx.useScopedContext;
var useTagProviderContext = ctx.useProviderContext;
var TagContextProvider = ctx.ContextProvider;
var TagScopedContextProvider = ctx.ScopedContextProvider;









exports.TagValueContext = TagValueContext; exports.TagRemoveIdContext = TagRemoveIdContext; exports.useTagContext = useTagContext; exports.useTagScopedContext = useTagScopedContext; exports.useTagProviderContext = useTagProviderContext; exports.TagContextProvider = TagContextProvider; exports.TagScopedContextProvider = TagScopedContextProvider;
