// @ariakit/react-core@0.4.18/cjs/__chunks/PH2DFCJS.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _ZVJRPAXYcjs = require('./ZVJRPAXY.cjs');



var _D6FV6EYScjs = require('./D6FV6EYS.cjs');


var _WULEED4Qcjs = require('./WULEED4Q.cjs');

// src/combobox/combobox-context.tsx
var _react = require('react');
var ComboboxListRoleContext = _react.createContext.call(void 0, 
  void 0
);
var ctx = _WULEED4Qcjs.createStoreContext.call(void 0, 
  [_ZVJRPAXYcjs.PopoverContextProvider, _D6FV6EYScjs.CompositeContextProvider],
  [_ZVJRPAXYcjs.PopoverScopedContextProvider, _D6FV6EYScjs.CompositeScopedContextProvider]
);
var useComboboxContext = ctx.useContext;
var useComboboxScopedContext = ctx.useScopedContext;
var useComboboxProviderContext = ctx.useProviderContext;
var ComboboxContextProvider = ctx.ContextProvider;
var ComboboxScopedContextProvider = ctx.ScopedContextProvider;
var ComboboxItemValueContext = _react.createContext.call(void 0, 
  void 0
);
var ComboboxItemCheckedContext = _react.createContext.call(void 0, false);










exports.ComboboxListRoleContext = ComboboxListRoleContext; exports.useComboboxContext = useComboboxContext; exports.useComboboxScopedContext = useComboboxScopedContext; exports.useComboboxProviderContext = useComboboxProviderContext; exports.ComboboxContextProvider = ComboboxContextProvider; exports.ComboboxScopedContextProvider = ComboboxScopedContextProvider; exports.ComboboxItemValueContext = ComboboxItemValueContext; exports.ComboboxItemCheckedContext = ComboboxItemCheckedContext;
