// @ariakit/react-core@0.4.18/cjs/__chunks/RDDLXR5T.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _EQD56WGFcjs = require('./EQD56WGF.cjs');




var _WULEED4Qcjs = require('./WULEED4Q.cjs');



var _7EQBAZ46cjs = require('./7EQBAZ46.cjs');

// src/focus-trap/focus-trap.tsx
var TagName = "span";
var useFocusTrap = _WULEED4Qcjs.createHook.call(void 0, 
  function useFocusTrap2(props) {
    props = _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {
      "data-focus-trap": "",
      tabIndex: 0,
      "aria-hidden": true
    }, props), {
      style: _7EQBAZ46cjs.__spreadValues.call(void 0, {
        // Prevents unintended scroll jumps.
        position: "fixed",
        top: 0,
        left: 0
      }, props.style)
    });
    props = _EQD56WGFcjs.useVisuallyHidden.call(void 0, props);
    return props;
  }
);
var FocusTrap = _WULEED4Qcjs.forwardRef.call(void 0, function FocusTrap2(props) {
  const htmlProps = useFocusTrap(props);
  return _WULEED4Qcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useFocusTrap = useFocusTrap; exports.FocusTrap = FocusTrap;
