// @ariakit/react-core@0.4.18/cjs/__chunks/VNVMHTCY.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _D6FV6EYScjs = require('./D6FV6EYS.cjs');


var _WULEED4Qcjs = require('./WULEED4Q.cjs');

// src/menubar/menubar-context.tsx
var _react = require('react');
var menubar = _WULEED4Qcjs.createStoreContext.call(void 0, 
  [_D6FV6EYScjs.CompositeContextProvider],
  [_D6FV6EYScjs.CompositeScopedContextProvider]
);
var useMenubarContext = menubar.useContext;
var useMenubarScopedContext = menubar.useScopedContext;
var useMenubarProviderContext = menubar.useProviderContext;
var MenubarContextProvider = menubar.ContextProvider;
var MenubarScopedContextProvider = menubar.ScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);








exports.useMenubarContext = useMenubarContext; exports.useMenubarScopedContext = useMenubarScopedContext; exports.useMenubarProviderContext = useMenubarProviderContext; exports.MenubarContextProvider = MenubarContextProvider; exports.MenubarScopedContextProvider = MenubarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
