// @ariakit/react-core@0.4.18/cjs/__chunks/YDPERDKF.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _KVXNVDJKcjs = require('./KVXNVDJK.cjs');


var _WULEED4Qcjs = require('./WULEED4Q.cjs');

// src/dialog/dialog-context.tsx
var _react = require('react');
var ctx = _WULEED4Qcjs.createStoreContext.call(void 0, 
  [_KVXNVDJKcjs.DisclosureContextProvider],
  [_KVXNVDJKcjs.DisclosureScopedContextProvider]
);
var useDialogContext = ctx.useContext;
var useDialogScopedContext = ctx.useScopedContext;
var useDialogProviderContext = ctx.useProviderContext;
var DialogContextProvider = ctx.ContextProvider;
var DialogScopedContextProvider = ctx.ScopedContextProvider;
var DialogHeadingContext = _react.createContext.call(void 0, void 0);
var DialogDescriptionContext = _react.createContext.call(void 0, void 0);









exports.useDialogContext = useDialogContext; exports.useDialogScopedContext = useDialogScopedContext; exports.useDialogProviderContext = useDialogProviderContext; exports.DialogContextProvider = DialogContextProvider; exports.DialogScopedContextProvider = DialogScopedContextProvider; exports.DialogHeadingContext = DialogHeadingContext; exports.DialogDescriptionContext = DialogDescriptionContext;
