// @medusajs/medusa@2.10.1/dist/api/admin/views/[entity]/columns/entity-mappings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ENTITY_MAPPINGS = void 0;
exports.ENTITY_MAPPINGS = {
    orders: {
        serviceName: "order",
        graphqlType: "Order",
        defaultVisibleFields: [
            "display_id",
            "created_at",
            "payment_status",
            "fulfillment_status",
            "total",
            "customer_display",
            "country",
            "sales_channel.name",
        ],
        fieldFilters: {
            // Fields that end with these suffixes will be excluded
            excludeSuffixes: ["_link"],
            // Fields that start with these prefixes will be excluded
            excludePrefixes: ["raw_"],
            // Specific field names to exclude
            excludeFields: ["order_change"],
        },
        computedColumns: {
            customer_display: {
                name: "Customer",
                computation_type: "customer_name",
                required_fields: [
                    "customer.first_name",
                    "customer.last_name",
                    "customer.email",
                ],
                optional_fields: ["customer.phone"],
                default_visible: true,
            },
            shipping_address_display: {
                name: "Shipping Address",
                computation_type: "address_summary",
                required_fields: [
                    "shipping_address.city",
                    "shipping_address.country_code",
                ],
                optional_fields: [
                    "shipping_address.address_1",
                    "shipping_address.province",
                    "shipping_address.postal_code",
                ],
                default_visible: false,
            },
            billing_address_display: {
                name: "Billing Address",
                computation_type: "address_summary",
                required_fields: [
                    "billing_address.city",
                    "billing_address.country_code",
                ],
                optional_fields: [
                    "billing_address.address_1",
                    "billing_address.province",
                    "billing_address.postal_code",
                ],
                default_visible: false,
            },
            country: {
                name: "Country",
                computation_type: "country_code",
                required_fields: ["shipping_address.country_code"],
                optional_fields: [],
                default_visible: true,
            },
        },
    },
    products: {
        serviceName: "product",
        graphqlType: "Product",
        defaultVisibleFields: [
            "title",
            "handle",
            "status",
            "created_at",
            "updated_at",
        ],
        fieldFilters: {
            excludeSuffixes: ["_link"],
            excludePrefixes: ["raw_"],
            excludeFields: [],
        },
        computedColumns: {},
    },
    customers: {
        serviceName: "customer",
        graphqlType: "Customer",
        defaultVisibleFields: [
            "email",
            "first_name",
            "last_name",
            "created_at",
            "updated_at",
        ],
        fieldFilters: {
            excludeSuffixes: ["_link"],
            excludePrefixes: ["raw_"],
            excludeFields: [],
        },
        computedColumns: {},
    },
    users: {
        serviceName: "user",
        graphqlType: "User",
        defaultVisibleFields: [
            "email",
            "first_name",
            "last_name",
            "created_at",
            "updated_at",
        ],
        fieldFilters: {
            excludeSuffixes: ["_link"],
            excludePrefixes: ["raw_"],
            excludeFields: [],
        },
        computedColumns: {},
    },
    regions: {
        serviceName: "region",
        graphqlType: "Region",
        defaultVisibleFields: ["name", "currency_code", "created_at", "updated_at"],
        fieldFilters: {
            excludeSuffixes: ["_link"],
            excludePrefixes: ["raw_"],
            excludeFields: [],
        },
        computedColumns: {},
    },
    "sales-channels": {
        serviceName: "salesChannel",
        graphqlType: "SalesChannel",
        defaultVisibleFields: [
            "name",
            "description",
            "is_disabled",
            "created_at",
            "updated_at",
        ],
        fieldFilters: {
            excludeSuffixes: ["_link"],
            excludePrefixes: ["raw_"],
            excludeFields: [],
        },
        computedColumns: {},
    },
};
